### PR TITLE
HTTP API doesn't understand HTTP auth creds in a URL

### PR DESCRIPTION
**Pull request by smn**
It should support `http://user:pass@host.com/`

### CODE REVIEW

File: go/apps/http_api_nostream/vumi_app.py [hunk 1]
```diff
@@ -123,10 +125,17 @@ def push(self, url, vumi_message):
...
         data = vumi_message.to_json().encode('utf-8')
         try:
+            auth, url = extract_auth_from_url(url.encode('utf-8'))
+            headers = {
+                'Content-Type': 'application/json; charset=utf-8',
+            }
+            if auth is not None:
+                headers.update({
+                    'Authorization': 'Basic %s' % (
+                        base64.b64encode('%s:%s' % auth))
```
Comment (justinvdm) [inline go/apps/http_api_nostream/vumi_app.py]: If `auth == ('u', None)`, we will be doing a `base64.b64encode('u:None')`. Just wondering if we should be handling that.

**Comment by justinvdm** [comment]
Looks good, just one small comment.

**Comment by smn** [comment]
@justinvdm ready for re-review.

**Comment by justinvdm** [comment]
:+1:

**Pull request merged**